modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Execution.scala

```diff
@@ -77,6 +77,10 @@ object Execution {
 sealed trait Result
 
 object Result {
-  case class OK[R](r: R) extends Result
+  case class OK(response: Response) extends Result
   case class Error[E](e: E) extends Result
+
+  sealed trait Response
+  case class Configured(r: String) extends Response
+  case class Observed(fileId: FileId) extends Response
```

Comment:
In the UI, you never have to look here, right? Ultimately, we need the id in `Step.fileId`, isn't it? I'm saying this because creating that case class would be much more problematic than updating the `fileId` directly in the `Step`.